OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitYoutubePlayerFragment.kt

```diff
@@ -0,0 +1,317 @@
+            }
+            val uri = Uri.parse(unit.data.encodedVideos.youtubeVideoInfo?.url)
+            /*
+             *  Youtube player loads the video using the video id from the url
+             *  the url has the following format "https://www.youtube.com/watch?v=3_yD_cEKoCk" where v is the video id
+             */
+            val videoId = uri.getQueryParameter("v")
+            if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+                youTubePlayer.toggleFullscreen()
+            }
+            videoId?.let {
+                youTubePlayer.loadVideo(
+                    it,
+                    (currentPos / AppConstants.MILLISECONDS_PER_SECOND).toFloat()
+                )
+            }
+        }
+    }
+
+    override fun canProcessSubtitles(): Boolean {
+        return youTubePlayer != null && currentPlayerState == PlayerState.PLAYING
+    }
+
+    override fun getPlayerCurrentPosition(): Long {
+        return if (youTubePlayer == null) 0 else currentTimeMillis
+    }
+
+    override fun setFullScreen(fullscreen: Boolean) {
+        if (DeviceSettingUtil.isDeviceRotationON(activity)) {
+            val orientation = resources.configuration.orientation
+            if ((isFullscreen && orientation == Configuration.ORIENTATION_LANDSCAPE) ||
+                (!isFullscreen && orientation != Configuration.ORIENTATION_LANDSCAPE)
+            ) {
+                return
+            }
+            youTubePlayer?.toggleFullscreen()
+        }
+    }
+
+    override fun playPauseVideoPlayback(pauseVideo: Boolean) {
+        if (pauseVideo) {
+            youTubePlayer?.pause()
+        } else {
+            youTubePlayer?.play()
+        }
+    }
+
+    private fun redirectToYoutubeDialog() {
+        releaseYoutubePlayer()
+        if (activity?.isDestroyed != true) {
+            (activity as BaseFragmentActivity)
+                .showAlertDialog(
+                    getString(R.string.assessment_unable_to_play_video),
+                    getString(R.string.assessment_unable_to_play_video_message),
+                    getString(R.string.assessment_open_on_youtube),
+                    { _: DialogInterface, _: Int ->
+                        BrowserUtil
+                            .open(
+                                activity,
+                                unit.data.encodedVideos.youtubeVideoInfo?.url, true
+                            )
+                    },
+                    getString(R.string.label_ok), null
+                )
+        }
+    }
+
+    private fun releaseYoutubePlayer() {
+        saveCurrentPlaybackPosition(currentTimeMillis)
+        (view?.findViewById<View>(R.id.player_container) as ViewGroup).removeAllViews()
+        youTubePlayerView?.release()
+        youTubePlayerView = null
+        youTubePlayer = null
+        activity?.let {
+            it.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_FULL_SENSOR
+        }
```

Comment:
Better to use: `activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_FULL_SENSOR
`